docs/source/tutorial/extract.rst

```diff
@@ -0,0 +1,432 @@
+            value_map(
+                in_col=9,  # age in hours (second)
+                m=lambda x: int(x) / 24,
+                out_col=CONCEPT.PHENOTYPE.EVENT_AGE_DAYS
+            ),
+            melt_map(
+                var_name=CONCEPT.PHENOTYPE.NAME,
+                map_for_vars={
+                    "EXTRA_EARDRUM": "Extra eardrum"
+                },
+                value_name=CONCEPT.PHENOTYPE.OBSERVED,
+                map_for_values=observed_yes_no
+            )
+        ]
+    ]
+
+Now let's break down how it works...
+
+Imports!
+--------
+
+.. code-block:: python
+
+    from kf_lib_data_ingest.common import constants
+    from kf_lib_data_ingest.etl.extract.operations import *
+    from kf_lib_data_ingest.etl.transform.standard_model.concept_schema import (
+        CONCEPT
+    )
+
+It's a Python module! Cool! That lets us do all kinds of neat stuff like
+import predefined constants and functions.
+
+Fetching the data
+-----------------
+
+.. code-block:: python
+
+    source_data_url = 'file://path/to/family_and_phenotype.tsv'
+
+The first thing that the extractor does for every config file is fetch the
+related source data. This specifies where the file lives so that the code can
+fetch it.
+
+Supported protocol prefixes are:
+``file://``, ``s3://``, ``http://``, ``https://``
+
+Loading the data
+----------------
+
+.. code-block:: python
+
+    source_data_loading_parameters = {}
+
+This file contains tab-separated values with a simple layout where the first
+row is the column headers, so it gets loaded correctly by default. With more
+complex files we'd have to define how to load the data. Those arguments would
+go here. For more information on custom loading read <TODO>.
+
+Extract operations
+------------------
+
+The operations list
+^^^^^^^^^^^^^^^^^^^
+
+.. code-block:: python
+
+    operations = [
+        ...
+    ]
+
+This is a list of special functions that implement all of the
+convert/unify/reshape/standardize stuff described earlier. The most useful
+functions are already written for you. You just have to invoke them
+appropriately.
+
+For more information about extract operation functions, read <TODO>.
+
+A value map operation with functional replacements
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. code-block:: python
+
+    value_map(
+        in_col="participant",
+        m={
+            r"PID(\d+)": lambda x: int(x),  # strip PID and 0-padding
+        },
+        out_col=CONCEPT.PARTICIPANT.ID
+    )
+
+This says "Use the ``participant`` column as input (in_col=), replace
+everything that matches the regular expression pattern ``^PID(\d+)$`` with just
+the captured part and remove the zero padding by running the captured part
+through the function ``lambda x: int(x)`` (m={...}), and then output the result
+to a ``CONCEPT.PARTICIPANT.ID`` column (out_col=)."
+
+The resulting intermediate output will look like:
+
+.. csv-table::
+    :header: "index", "<CONCEPT.PARTICIPANT.ID>"
```

Comment:
Why are all of the mapped concept col names enclosed in <>?

Reply:
. vs | stylistic decision

Reply:
Oh I didn't even notice the delimiter 😄 , I was just talking about the <>

Reply:
I used < because I used . and not |

Reply:
it's arbitrary